src/components/Sidebar/ActionNode.tsx

```diff
@@ -1,49 +1,212 @@
 import React, { FC, memo, useCallback, ChangeEvent } from 'react';
 import { SidebarButton } from '.';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faPlus } from '@fortawesome/free-solid-svg-icons';
-
+import { faPlus, faCheckCircle } from '@fortawesome/free-solid-svg-icons';
+import { setStateAction } from 'utils/builder';
 import DropDown from 'components/elements/DropDown';
-import { Pathway, State } from 'pathways-model';
+import Input from 'components/elements/Input';
+import { Pathway, GuidanceState, Action } from 'pathways-model';
 import useStyles from './styles';
+import shortid from 'shortid';
 
 const nodeTypeOptions = [
   { label: 'Action', value: 'action' },
   { label: 'Branch', value: 'branch' }
 ];
 
+const actionTypeOptions = [
+  { label: 'Medication Request', value: 'Medication Request' },
+  { label: 'Service Request', value: 'Service Request' },
+  { label: 'Careplan', value: 'Careplan' }
+];
+
+const codeSystemOptions = [
+  { label: 'RXNORM', value: 'http://www.nlm.nih.gov/research/umls/rxnorm' },
+  { label: 'SNOMED', value: 'http://snomed.info/sct' },
+  { label: 'LOINC', value: ' http://loinc.org' }
+];
+
 interface ActionNodeProps {
   pathway: Pathway;
-  currentNode: State;
+  currentNode: GuidanceState;
   changeNodeType: (event: string) => void;
   addNode: (event: string) => void;
+  updatePathway: (pathway: Pathway) => void;
 }
 
-const ActionNode: FC<ActionNodeProps> = ({ pathway, currentNode, changeNodeType, addNode }) => {
+const ActionNode: FC<ActionNodeProps> = ({
+  pathway,
+  currentNode,
+  changeNodeType,
+  addNode,
+  updatePathway
+}) => {
   const styles = useStyles();
   const selectNodeType = useCallback(
     (event: ChangeEvent<{ value: string }>): void => {
       changeNodeType(event?.target.value || '');
     },
     [changeNodeType]
   );
+  const currentNodeKey = currentNode?.key;
```

Comment:
We should be consistent with using `currentNode` and its key:
1. For starters I don't think `currentNode` can be null so we shouldn't need to check for that
2. This also means all of the `if (!currentNodeKey) return;` statements can be removed
3. The `currentNode` is already passed as a prop but in multiple places you use the key to find the node from the pathway instead of just using the node passed in.
4. Some places use `currentNodeKey` but others use `currentNode.key`. I don't really mind which one we use but it should be consistent.

Reply:
currentNodeKey can be undefined, so we still have to check it.  Other stuff I agree.